src/ctelnet.cpp

```diff
@@ -1627,9 +1634,9 @@ void cTelnet::processTelnetCommand(const std::string& command)
                 mpHost->updateProxySettings(mpDownloader);
                 auto request = QNetworkRequest(QUrl(url));
                 mudlet::self()->setNetworkRequestDefaults(url, request);
-                QNetworkReply* reply = mpDownloader->get(request);
+                packageDownloadReply = mpDownloader->get(request);
                 mpProgressDialog = new QProgressDialog(tr("downloading game GUI from server"), tr("Cancel", "Cancel download of GUI package from Server"), 0, 4000000, mpHost->mpConsole);
-                connect(reply, &QNetworkReply::downloadProgress, this, &cTelnet::setDownloadProgress);
+                connect(packageDownloadReply, &QNetworkReply::downloadProgress, this, &cTelnet::setDownloadProgress);
```

Comment:
You need the following I think to wire the cancel button up to the (misnamed IMVHO - should be `slot_interfaceDownloadCancelled`) **SLOT** method `(void) cTelnet::interfaceDownloadCancelled()` - something along the lines of:
```cpp
    connect(mpProgressDialog, &QProgressDialog::canceled, this, cTelnet::interfaceDownloadCancelled);
```